pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ExecutorServiceUtils.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.executor;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.apache.pinot.common.utils.NamedThreadFactory;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ExecutorServiceUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ExecutorServiceUtils.class);
+
+  private ExecutorServiceUtils() {
+  }
+
+  public static ExecutorService createDefault(String baseName) {
+    return Executors.newCachedThreadPool(new NamedThreadFactory(baseName));
+  }
+
+  public static ExecutorService create(PinotConfiguration conf, String confPrefix, String baseName) {
+    //TODO: make this configurable
+    return Executors.newCachedThreadPool(new NamedThreadFactory(baseName));
+  }
```

Comment:
The idea here is that in Java 21 we may want to use a virtual pool here. 

That can be disabled by default (therefore we would need to set a property in order to use loom) or enabled by default (in which case we would use a property to do not use loom). Given that these executors are created in different parts of the code, this util will be used to read the properties in a single place.

One thing I don't like about the current approach is that `create` and `createDefault` are quite generic. Should be something like `createUnbound` because the idea is that we should call this method whenever we want to create a unbound thread pool executor

Reply:
Here you have an example on how to make executors customizable: https://github.com/gortiz/pinot/pull/6

We can merge it here if you want, although it may be better to do that in another PR to keep this one simpler